### PR TITLE
EZP-32103: Add Varnish integration tests

### PR DESCRIPTION
**Pull request by mnocon**
JIRA: https://issues.ibexa.co/browse/EZP-32103

Changes required by https://github.com/ezsystems/ezplatform-http-cache/pull/138

Couple of things here:
1) Added a new Context (TimeContext) that allows to wait a couple of seconds
2) Added a check that the login action is successful - we should be redirected after the login action to another page
3) As I've played with the Chromium a bit I've noticed that the format for HTTP headers is different - I've decided to include support for it in this PR 😄 
4) And the most important change: HTTP flushing. Here's a glimpse into our HTTP cache:
When we want to purge cache: https://github.com/ezsystems/ezplatform-http-cache/blob/master/src/PurgeClient/VarnishPurgeClient.php#L27
we call the `invalidateTags` method. We're using the `PURGEKEYS` way, so this code is relevant:
https://github.com/FriendsOfSymfony/FOSHttpCache/blob/master/src/ProxyClient/Varnish.php#L197-L205
The most important part: this part does not purge the cache 🙃  It simply pushes the request to purge the cache to a queue which is emptied (flushed) in InvalidationListener:
https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/master/src/EventListener/InvalidationListener.php

The cache is purged only when the requests are sent (the queue is flushed).

The InvalidationListener is an EventListener which listens to three events:
```
    public static function getSubscribedEvents()
    {
        return [
            KernelEvents::TERMINATE => 'onKernelTerminate',
            KernelEvents::EXCEPTION => 'onKernelException',
            ConsoleEvents::TERMINATE => 'onConsoleTerminate',
        ];
    }
```

Sadly for us, running our API from Behat triggers none of the above. That's why I've decided to flush the Queue manually when operations involving Content are executed - I'm not 100% happy with this (this can be really error prone, as it's easy to forget this). We might need to revisit it in the future if BehatBundle continues to grow - maybe it should be added as an AfterStep hook (but then it would trigger for all Steps, which might be an overkill/slight performance issue - something to assess in the future).

### CODE REVIEW

**Review by micszo — APPROVED** [review]
> to flush the Queue manually

Maybe it's worth adding this approach to some sort of internal "doc", not sure where though. 😁

**Review by bogusez — APPROVED** [review]

**Comment by mnocon** [comment]
> Maybe it's worth adding this approach to some sort of internal "doc", not sure where though. 😁
I'd just keep it in mind when writing code/reviewing, that should be enough for now.

File: src/lib/API/Facade/ContentFacade.php [hunk 1]
```diff
@@ -28,12 +29,20 @@ class ContentFacade
     /** @var \EzSystems\Behat\API\ContentData\ContentDataProvider */
     private $contentDataProvider;
 
-    public function __construct(ContentService $contentService, LocationService $locationService, URLAliasService $urlAliasService, ContentDataProvider $contentDataProvider)
+    /** @var \FOS\HttpCacheBundle\CacheManager */
+    private $cacheManager;
+
+    public function __construct(ContentService $contentService,
```
Comment (crevillo) [inline src/lib/API/Facade/ContentFacade.php:35]: supernitpicking: is this the code style we follow for this kind of constructors? I think i rember them being more like
```
public function __construct(
    ContentService $contentService,
   LocationService $locationService,
   ... 
) {
```

Comment (mnocon) [reply src/lib/API/Facade/ContentFacade.php:35]: I did a quick search for current usages and you're right! Not sure right now why I opted for this version, probably I'm too used to cs-fixer taking care of everything.

Wonder if there is a setting that can enforce it somehow...

Comment (mnocon) [reply src/lib/API/Facade/ContentFacade.php:35]: Maybe https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.16/doc/rules/function_notation/method_argument_space.rst , I will have a look at it tomorrow 😄

**Pull request merged**